### ISSUE TITLE
Timer subscription doesn't fire if subscriptions is changed through task

### ISSUE THREAD

**Issue by ahnlabb** [elm/core#993]
Likely a duplicate of #980.
# SSCCE (https://ellie-app.com/3r9VW4dDR3Ba1)
``` elm
import Browser
import Html exposing (..)
import Html.Attributes exposing (..)
import Task
import Time


main =
    Browser.element
        { init = init
        , view = view
        , update = update
        , subscriptions = subscriptions
        }


type Model
    = Loading
    | Loaded Page


type alias Page =
    { time : Time.Posix
    }


init : () -> ( Model, Cmd Msg )
init _ =
    ( Loading
    , Task.perform Tick Time.now
    )



-- UPDATE


type Msg
    = Tick Time.Posix


update : Msg -> Model -> ( Model, Cmd Msg )
update msg model =
    case ( msg, model ) of
        ( Tick newTime, Loading ) ->
            ( Loaded (Page newTime)
            , Cmd.none
            )

        ( Tick newTime, Loaded page ) ->
            ( Loaded { page | time = newTime }
            , Cmd.none
            )



-- SUBSCRIPTIONS


subscriptions : Model -> Sub Msg
subscriptions model =
    case model of
        Loaded _ ->
            Time.every 1000 Tick

        Loading ->
            Sub.none



-- VIEW


view : Model -> Html Msg
view model =
    case model of
        Loaded page ->
            viewPage page

        _ ->
            div [] []


viewPage : Page -> Html Msg
viewPage { time } =
    let
        zone =
            Time.utc

        toPaddedInt =
            (String.padLeft 2 '0') << String.fromInt

        hour =
            toPaddedInt (Time.toHour zone time)

        minute =
            toPaddedInt (Time.toMinute zone time)

        second =
            toPaddedInt (Time.toSecond zone time)
    in
        h1 [] [ text (hour ++ ":" ++ minute ++ ":" ++ second) ]
```

## Expected Behavior
The page should display the UTC time, updating every second.

## Actual Behavior
The page displays the UTC time when the page loads but after that does not update.

## Notes
The `subscriptions` function is evaluated once to the value `Time.every 1000 Tick` (confirmed with `Debug.log`) however the runtime does not send any messages after that.

**Comment by ohanhi** [comment]
A workaround for this issue is changing the initial Cmd to:
```
Process.sleep 1 |> Task.andThen (\_ -> Time.now) |> Task.perform Tick
```

This issue just came up on the Beginners channel on Slack. I came up with the theory that the `Time.now` resolves too quickly, and the subscriptions have not been properly set up at that point. Based on this assumption, I figured adding a tiny delay should be enough to push the Msg to the next tick, when the subscriptions have probably been set up to listen to changes in the model. This does seem to work.

Here's the Ellie from above, with the workaround mentioned: https://ellie-app.com/4x47XRpLgbra1

**Comment by ahnlabb** [comment]
Thank you @ohanhi! That is very interesting and nice to have a workaround. Then this would be unrelated to #980?

**Comment by ohanhi** [comment]
@ahnlabb I can't be sure on that, but based on my working theory these could be separate issues. My working theory might be flawed, though.

**Comment by evancz** [comment]
I tried this SSCCE with the changes in https://github.com/elm/core/commit/704dcc08d4617d45c8b0fce7a68729335c880756 and it looks like it works as expected.

Thank you for reporting a fresh issue and SSCCE! The fix should come out with an elm/core 1.0.3 release.
